feature_engine/variable_manipulation.py

```diff
@@ -2,33 +2,77 @@
 # License: BSD 3 clause
 # functions shared across transformers
 
+from typing import List, Optional, Union
 
-def _define_variables(variables):
-    # Check that variable names are passed in a list.
-    # Can take None as value
+import pandas as pd
+
+
+def _define_variables(variables: Union[None, List[str]]) -> Union[List[str], None]:
+    """
+    Takes string or list of strings and checks if argument is list of strings.
+    Can take None as argument.
+
+    Args:
+        variables: string or list of strings
+
+    Returns:
+        List of strings
+    """
+
+    #! Can take Tuple of variables and return list of tuple. Is it accaptable?
     if not variables or isinstance(variables, list):
         variables = variables
     else:
         variables = [variables]
+
     return variables
 
 
-def _define_numerical_dict(the_dict):
-    # Check that the entered dictionary is indeed a dictionary of integers and floats
-    # Can take None as value
+def _define_numerical_dict(the_dict: Union[Optional[str], dict]) -> Union[Optional[str], dict]:
+    """
+    Takes dictionary and checks if all values in dictionary are integers and floats.
+    Can take None as argument.
+
+    Args:
+        the_dict: Dict to perform check against
+
+    Raises:
+        ValueError: If all values of dict are not int or float
+        TypeError: When argument type is not dict
+
+    Returns:
+        None or the dict
+    """
+
     if not the_dict:
         the_dict = the_dict
     elif isinstance(the_dict, dict):
         if not all([isinstance(x, (float, int)) for x in the_dict.values()]):
             raise ValueError('All values in the dictionary must be integer or float')
     else:
         raise TypeError('The parameter can only take a dictionary or None')
+
     return the_dict
 
 
-def _find_numerical_variables(X, variables=None):
-    # Find numerical variables in a data set or check that
-    # the variables entered by the user are numerical.
+def _find_numerical_variables(X: pd.DataFrame, variables: List[str]=None) -> List[str]:
+    """
+    Takes Pandas DataFrame along with list of colum names and
+    checks if these columns are numerical. If not provided columns,
+    checks if all of them are numerical.
```

Comment:
This function checks that variables entered by user are of type numerical. If no variables were entered by the user, it will capture in a list all variables of type numerical present in the dataset.
Not sure this is clear from this docstring. Could we re-word a bit?